include/swift/AST/DiagnosticsSema.def

```diff
@@ -2869,9 +2869,9 @@ ERROR(compiler_evaluable_ref_non_compiler_evaluable,none,
       "@compilerEvaluable functions may not reference non-@compilerEvaluable functions", ())
 
 // @noDerivative attribute
-ERROR(noderivative_only_on_stored_properties_in_differentiable_structs,none,
-      "'@noDerivative' is only allowed on stored properties in structure types "
-      "that declare a conformance to 'Differentiable'", ())
+ERROR(noderivative_only_on_stored_properties_in_differentiable_types,none,
```

Comment:
This is a weird name. Consider being explicit about structs and classes. Shortening can be done if you change "stored_properties" to "fields".
```suggestion
ERROR(noderivative_only_on_differentiable_structs_or_class_fields,none,
```

Reply:
Done in https://github.com/apple/swift/pull/25914/commits/9182edaa6938200b72facd2e7359e084ff808223.